python_modules/dagster/dagster/core/object_store.py

```diff
@@ -89,13 +89,39 @@ def set_value(self, obj, context, runtime_type, paths):
             return self.TYPE_REGISTRY[runtime_type.name].set_object(
                 self, obj, context, runtime_type, paths
             )
+        elif runtime_type.name is None:
+            composite_overrides = {
+                t.name in self.TYPE_REGISTRY
+                for t in runtime_type.inner_types
+                if t.name in self.TYPE_REGISTRY
+            }
+            if composite_overrides:
+                check.not_implemented(
+                    'Don\'t know how to store a composite object with inner type storage '
```

Comment:
How about:

`You are attempting to store type {type_name} in a object store. It has specialized storage behavior (configured via the TYPE_STORAGE_PLUGIN_REGISTRY), we do not support storing Nullable or Lists of types with customized storage at the moment. See issue <<num>>  for details`

Reply:
👍